anvil/core/src/eth/transaction/mod.rs

```diff
@@ -984,12 +984,20 @@ impl PendingTransaction {
     /// Creates a new pending transaction and tries to verify transaction and recover sender.
     pub fn new(transaction: TypedTransaction) -> Result<Self, SignatureError> {
         let sender = transaction.recover()?;
-        Ok(Self::with_sender(transaction, sender))
+        Ok(Self { hash: transaction.hash(), transaction, sender })
     }
 
-    /// Creates a new transaction with the given sender
-    pub fn with_sender(transaction: TypedTransaction, sender: Address) -> Self {
-        Self { hash: transaction.hash(), transaction, sender }
+    /// Creates a new transaction with the given sender.
+    ///
+    /// In order to prevent collisions from multiple different impersonated accounts, we update the
+    /// transaction's hash with the address to make it unique.
+    ///
+    /// See: <https://github.com/foundry-rs/foundry/issues/3759>
+    pub fn with_impersonated(transaction: TypedTransaction, sender: Address) -> Self {
```

Comment:
so, formally the hash of the impersonated transaction is `keccak256(rlp(tx) . sender)` where `.` is concatenation, correct?

Reply:
yes, this makes it unique